src/peft/peft_model.py

```diff
@@ -163,13 +163,35 @@ def peft_config(self, value: dict[str, PeftConfig]):
         else:
             self.base_model.peft_config = value
 
+    def subtract_pissa_init(self, pissa_initial_dir, output_state_dict, kwargs):
+        adapter_name = os.path.basename(pissa_initial_dir)
+        self.load_adapter(os.path.dirname(pissa_initial_dir), adapter_name=adapter_name, subfolder=adapter_name)
+        pissa_init_state_dict = get_peft_model_state_dict(
+            self,
+            state_dict=kwargs.get("state_dict", None),
+            adapter_name=adapter_name,
+        )
+        tensors_lora = {}
+        for name in pissa_init_state_dict.keys():
+            ## W = W^{res} + A_0 \times B_0,
+            ## W + \Delta W = W^{res} + A \times B,
+            ## \Delta W = A \times B - A_0 \times B_0 = [A | A_0] \times [B | B_0]^T = A'B'.
+            tensors_lora[name] = (
+                torch.cat([output_state_dict[name], pissa_init_state_dict[name]], dim=0)
+                if "lora_A" in name
+                else torch.cat([output_state_dict[name], -pissa_init_state_dict[name]], dim=1)
+            )
+        self.delete_adapter(adapter_name)
```

Comment:
Hmm, we delete the adapter so that we don't save it twice, or why is that? Ideally, I would like to avoid this. IIUC, right now, when a user calls `save_pretrained(..., save_as_lora=True)`, afterwards, the adapters are all deleted, which would be quite surprising.

Reply:
The deleted adapter is the newly injected `pissa_init` adapter. I have modified the variable name from `adapter_name` to `initial_adapter`.